proxy.go

```diff
@@ -327,22 +334,20 @@ func (rp *reverseProxy) serveFromCache(s *scope, srw *statResponseWriter, req *h
 		if err = userCache.Fail(key); err != nil {
 			log.Errorf("%s: %s; query: %q", s, err, q)
 		}
-		err = RespondWithoutData(srw)
+
+		err = RespondWithData(srw, reader, contentMetadata, 0*time.Second, bufferedRespWriter.StatusCode())
```

Comment:
I don't know how it was before the redis cache PR  but shouldn't we respond a http.StatusInternalServerError (i.e 500) instead of a bufferedRespWriter.StatusCode()?
The error from CH will still be written in the content of the http body.
I played at bit with CH http endpoint and it seems to return either 400 or 500

Reply:
If we do so, we won't propagate real status code from Clickhouse and I believe it's what is desired. Yes the error payload will be there regardless

Reply:
Ok